ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -349,6 +350,20 @@
 
   private final AtomicLong sparkSessionId = new AtomicLong();
 
+  private final Map<String, QueryState> queryStateMap = new HashMap<>();
+
+  public Object getQueryState(String queryId) {
```

Comment:
Can we update the return type too so we don't need any casting?

Reply:
Sure!